modules/core/src/smithy4s/Transformation.scala

```diff
@@ -119,4 +119,30 @@ object Transformation {
       }
     }
 
+  implicit def mappedHintsServiceTransform[Alg[_[_, _, _, _, _]]]: Transformation[Hints=>Hints, Service[Alg], Service[Alg]] =
+    (func: Hints => Hints, that: Service[Alg]) => new Service[Alg] {
+      override type Operation[I, E, O, SI, SO] = that.Operation[I, E, O, SI, SO]
+
+      override val endpoints: List[Endpoint[_, _, _, _, _]] = that.endpoints.map(_.mapHints(func))
+
+      override def endpoint[I, E, O, SI, SO](op: Operation[I, E, O, SI, SO]): (I, Endpoint[I, E, O, SI, SO]) = {
+        that.endpoint(op) match {
+          case (i, endpoint) => (i, endpoints.find(_.id == endpoint.id).get.asInstanceOf[Endpoint[I, E, O, SI, SO]])
```

Comment:
Nearly there. What I'd do would be back it with a map instead of list. 

So I'd have : 

```scala
private val endpointMap : Map[ShapeId, Endpoint[...]] = ??? 
def endpoints: List[Endpoint[...]] = endpointMap.values.toList 
```

and then use the map for a quicker access.